### ISSUE TITLE
Questions about inference

### ISSUE THREAD

**Issue by tony-laoshi** [gangweiX/ACVNet#18]
First of all, thank you very much for your very good work！！！

I tried to add an **inference()** function to **main.py** to check the inference effect. But the result is very poor, the loss value is over 30. The strange thing is that when I use your **train()** function for training, it works very well on both the training set and the test set (validation as I understand it). 

I have done the problem locating:
- I combed through the data import for processing and didn't find anything wrong.
- Checked that model was set to **model.eval()**. Yes, I set 

I would like to ask if there is a script for inference? I hope I can get some help. 

Here is my function that performs the inference. I have not made changes to everything else
```
      def inference():
      for epoch_idx in range(start_epoch, args.epochs):
          adjust_learning_rate(optimizer, epoch_idx, args.lr, args.lrepochs)
          # training
          # for batch_idx, sample in enumerate(TrainImgLoader):
          #     global_step = len(TrainImgLoader) * epoch_idx + batch_idx
          #     start_time = time.time()
          #     do_summary = global_step % args.summary_freq == 0
          #     loss, scalar_outputs, image_outputs = train_sample(sample, compute_metrics=do_summary)
          #     if do_summary:
          #         save_scalars(logger, 'train', scalar_outputs, global_step)
          #         save_images(logger, 'train', image_outputs, global_step)
          #     del scalar_outputs, image_outputs
          #     print('Epoch {}/{}, Iter {}/{}, train loss = {:.3f}, time = {:.3f}'.format(epoch_idx, args.epochs,
          #                                                                                batch_idx,
          #                                                                                len(TrainImgLoader), loss,
          #                                                                                time.time() - start_time))
          
          # # saving checkpoints
  
          # if (epoch_idx + 1) % args.save_freq == 0:
          #     checkpoint_data = {'epoch': epoch_idx, 'model': model.state_dict(), 'optimizer': optimizer.state_dict()}
          #     #id_epoch = (epoch_idx + 1) % 100
          #     torch.save(checkpoint_data, "{}/checkpoint_{:0>6}.ckpt".format(args.logdir, epoch_idx))
          # gc.collect()
  
          if (epoch_idx) % 1 == 0:
  
          # # testing
              avg_test_scalars = AverageMeterDict()
              for batch_idx, sample in enumerate(TestImgLoader):    
                  global_step = len(TestImgLoader) * epoch_idx + batch_idx
                  start_time = time.time()
                  do_summary = global_step % args.summary_freq == 0
                  loss, scalar_outputs, image_outputs = test_sample(sample, compute_metrics=do_summary)
                  if do_summary:
                     save_scalars(logger, 'test', scalar_outputs, global_step)
                     save_images(logger, 'test', image_outputs, global_step)
                  avg_test_scalars.update(scalar_outputs)
                  del scalar_outputs, image_outputs
                  print('Epoch {}/{}, Iter {}/{}, test loss = {:.3f}, time = {:3f}'.format(epoch_idx, args.epochs,
                                                                                           batch_idx,
                                                                                           len(TestImgLoader), loss,
                                                                                           time.time() - start_time))
              avg_test_scalars = avg_test_scalars.mean()
              # save_scalars(logger, 'fulltest', avg_test_scalars, len(TrainImgLoader) * (epoch_idx + 1))
              print("avg_test_scalars", avg_test_scalars)
              gc.collect()`
```

The run command is:
`
python main.py --loadckpt finetune_results/checkpoint_000499.ckpt --epochs 1 --logdir inference_results --test_batch_size 1 --summary_freq 1
`
